pyhive/hive.py

```diff
@@ -49,6 +49,45 @@
 }
 
 
+def get_sasl_client(host, sasl_auth, service=None, username=None, password=None):
+    import sasl
+    sasl_client = sasl.Client()
+    sasl_client.setAttr('host', host)
+
+    if sasl_auth == 'GSSAPI':
+        sasl_client.setAttr('service', service)
+    elif sasl_auth == 'PLAIN':
+        sasl_client.setAttr('username', username)
+        sasl_client.setAttr('password', password)
+    else:
+        raise AssertionError
+
+    sasl_client.init()
+    return sasl_client
+
+
+def get_pure_sasl_client(host, sasl_auth, service=None, username=None, password=None):
+    from pyhive.sasl_compat import PureSASLClient
+
+    if sasl_auth == 'GSSAPI':
```

Comment:
let's move it into get_sasl_kwargs function, looks the same as above

Reply:
@bkyryliuk [setAttr](https://github.com/cloudera/python-sasl/blob/master/sasl/saslwrapper.pyx#L53) is a method of python-sasl library which is used to set client attributes one at a time. 
whereas pure-sasl client accepts **kwargs in dictionary form. I am not sure how to combine both behaviors in a single function. 
I think we should keep it as it is.

Reply:
ah, missed that, good point. It is still doable: https://stackoverflow.com/questions/30212361/setting-multiple-object-attributes-at-once
but probably is not worthwhile